app/Helpers/DistanceHelper.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace App\Helpers;
+
+use App\Models\User;
+
+class DistanceHelper
+{
+    /**
+     * Format the distance according to the preferences of the user.
+     *
+     * @param  User  $user
+     * @param  int  $distance
+     * @param  string  $unit
+     * @return string
+     */
+    public static function format(User $user, int $distance, string $unit): string
+    {
+        if ($user->distance_format === User::DISTANCE_UNIT_KM) {
+            if ($unit === User::DISTANCE_UNIT_MILES) {
+                $distance = round($distance * 1.609344, 2);
+            }
+        }
+
+        if ($user->distance_format === User::DISTANCE_UNIT_MILES) {
+            if ($unit === User::DISTANCE_UNIT_KM) {
+                $distance = round($distance / 1.609344, 2);
+            }
+        }
```

Comment:
```suggestion
        if ($user->distance_format === User::DISTANCE_UNIT_KM && $unit === User::DISTANCE_UNIT_MILES) {
            $distance = round($distance * 1.609344, 2);
        } else if ($user->distance_format === User::DISTANCE_UNIT_MILES && $unit === User::DISTANCE_UNIT_KM) {
            $distance = round($distance / 1.609344, 2);
        }
```

Reply:
the `distance` is a number (integer). so when converting it to the other unit, shouldn't it converted to an integer as well (like doing a round with 0 precision instead of 2) ?

Reply:
@asbiin the thing is, if we do that, we'll lose a great precision, because 3km = 1.86 miles, not 2 miles. And 2 miles is much farther than 1.8miles. So...